### PR TITLE
add MEModel

### PR DESCRIPTION
**Pull request by AurelienJaquier**
also small refactoring for `EModel`, `search_pdfs` and `validate`

Also take into account all possible legacy checkpoint paths, that depend on `EModelMetadata.as_string()`, that has been updated several times

### CODE REVIEW

**Comment by AurelienJaquier** [comment]
Also take into account all possible legacy checkpoint paths, that depend on EModelMetadata.as_string, that has been updated several times

**Review by adrien-berchet — COMMENTED** [review]
Just minor comments

File: bluepyemodel/tools/utils.py [hunk 1]
```diff
@@ -28,16 +28,51 @@
...
+    """
+    checkpoint_path = Path(checkpoint_path)
+    return (
+        checkpoint_path.is_file()
+        or checkpoint_path.with_suffix(checkpoint_path.suffix + ".tmp").is_file()
+    )
+
 
+def get_checkpoint_path(metadata, seed=None):
+    """Get checkpoint path. Use legacy format if any is found, if not use latest format."""
```
[the remark this answers is not in the document]
Comment (AurelienJaquier) [reply bluepyemodel/tools/utils.py]: fixed in latest commit

File: bluepyemodel/tools/utils.py [hunk 2]
```diff
@@ -74,60 +109,18 @@ def yesno(question):
...
-    ]:
-        search_str = f"{field}="
-        checkpoint_metadata[field] = next(
-            (e.replace(search_str, "") for e in filename if search_str in e), None
-        )
+    search_str = "seed="
+    seed = next((e.replace(search_str, "") for e in filename if search_str in e), None)
 
-    return checkpoint_metadata
+    return int(seed)
```
[the remark this answers is not in the document]
Comment (AurelienJaquier) [reply bluepyemodel/tools/utils.py:123]: Good catch! I changed default seed to be 0 if it is not found in the filename

File: tests/unit_tests/test_optimisation.py [hunk 3]
```diff
@@ -14,43 +14,51 @@
...
-from bluepyemodel.tools.utils import parse_checkpoint_path
+from bluepyemodel.tools.utils import get_seed_from_checkpoint_path
 from bluepyemodel.emodel_pipeline.emodel_metadata import EModelMetadata
 
 
-def test_get_checkpoint_path():
-    metadata = EModelMetadata(emodel="L5PC", ttype="t type", iteration_tag="test")
+def checkpoint_check(dir, fname, metadata, inner_dir):
+    f = dir / fname
+    f.write_text("")
```
[the remark this answers is not in the document]
Comment (AurelienJaquier) [reply tests/unit_tests/test_optimisation.py]: Thanks, changed in latest commit

**Review by adrien-berchet — APPROVED** [review]

**Pull request merged**